pylint/lint/pylinter.py

```diff
@@ -1004,23 +1005,22 @@ def _check_files(self, get_ast, file_descrs):
                     msg = get_fatal_error_message(filepath, template_path)
                     if isinstance(ex, AstroidError):
                         symbol = "astroid-error"
-                        msg = (filepath, msg)
+                        self.add_message(symbol, args=(filepath, msg))
                     else:
                         symbol = "fatal"
-                    self.add_message(symbol, args=msg)
+                        self.add_message(symbol, args=msg)
 
-    def _check_file(self, get_ast, check_astroid_module, name, filepath, modname):
+    def _check_file(self, get_ast, check_astroid_module, file: FileItem):
         """Check a file using the passed utility functions (get_ast and check_astroid_module)
 
         :param callable get_ast: callable returning AST from defined file taking the following arguments
         - filepath: path to the file to check
         - name: Python module name
         :param callable check_astroid_module: callable checking an AST taking the following arguments
         - ast: AST of the module
-        :param str name: full name of the module
-        :param str filepath: path to checked file
-        :param str modname: name of the checked Python module
+        :param FileItem file: data about the file
         """
+        name, filepath, modname = file
```

Comment:
Why do a tuple unpacking. With `NamedTuple` you can access the attributes by name.
It might even make sense to just pass `file` to `set_current_module` / `get_ast` and do the attribute access there. A bit more refactoring though.

Reply:
I left the refactor for another time, since this is becoming quite a large PR.

One thing I noticed is some ambiguity between `modname` and `modpath`. They seem to be used for the same attribute. We currently use `modpath`, is that okay? Or do you want to start using `modname`?

Reply:
+1 for keeping the refactors for later :)